File: experiments/runs/2026-09-16_08-12-47_claim-office-example-mapping_exact-sol-v1.6-test-list-dimensions-cc_opus-5-no-thinking/src/price-list.ts
/** The MHPCO price list: published insurance value and base premium per item type. */
const SWORD_BASE_PREMIUM_IN_G = 100;
const AMULET_BASE_PREMIUM_IN_G = 60;
const STAFF_BASE_PREMIUM_IN_G = 80;
const POTION_BASE_PREMIUM_IN_G = 40;
const COMPONENT_BASE_PREMIUM_IN_G = 25;

const SWORD_INSURANCE_VALUE_IN_G = 1000;
const AMULET_INSURANCE_VALUE_IN_G = 600;
const STAFF_INSURANCE_VALUE_IN_G = 800;
const POTION_INSURANCE_VALUE_IN_G = 400;
const COMPONENT_INSURANCE_VALUE_IN_G = 250;

const BASE_PREMIUM_IN_G: Record<string, number> = {
  sword: SWORD_BASE_PREMIUM_IN_G,
  amulet: AMULET_BASE_PREMIUM_IN_G,
  staff: STAFF_BASE_PREMIUM_IN_G,
  potion: POTION_BASE_PREMIUM_IN_G,
  rune: COMPONENT_BASE_PREMIUM_IN_G,
  moonstone: COMPONENT_BASE_PREMIUM_IN_G,
};

const INSURANCE_VALUE_IN_G: Record<string, number> = {
  sword: SWORD_INSURANCE_VALUE_IN_G,
  amulet: AMULET_INSURANCE_VALUE_IN_G,
  staff: STAFF_INSURANCE_VALUE_IN_G,
  potion: POTION_INSURANCE_VALUE_IN_G,
  rune: COMPONENT_INSURANCE_VALUE_IN_G,
  moonstone: COMPONENT_INSURANCE_VALUE_IN_G,
};

const COMPONENT_TYPES = ["rune", "moonstone"];

export function basePremiumOf(itemType: string): number {
  const basePremium = BASE_PREMIUM_IN_G[itemType];
  if (basePremium === undefined) throw new Error(`Unknown item type: ${itemType}`);
  return basePremium;
}

export function isComponent(itemType: string): boolean {
  return COMPONENT_TYPES.includes(itemType);
}

export function insuranceValueOf(itemType: string): number {
  const insuranceValue = INSURANCE_VALUE_IN_G[itemType];
  if (insuranceValue === undefined) throw new Error(`Unknown item type: ${itemType}`);
  return insuranceValue;
}

File: experiments/runs/2026-09-20_07-18-54_claim-office-java-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/main/java/PriceList.java
import java.util.Map;

/** The MHPCO price list: insurance values and base premiums per item type. */
public final class PriceList {

    private static final Map<String, Integer> INSURANCE_VALUES = Map.of(
            "sword", 1000,
            "amulet", 600,
            "staff", 800,
            "potion", 400,
            "rune", 250,
            "moonstone", 250);

    private static final Map<String, Integer> BASE_PREMIUMS = Map.of(
            "sword", 100,
            "amulet", 60,
            "staff", 80,
            "potion", 40,
            "rune", 25,
            "moonstone", 25);

    private PriceList() {
    }

    public static boolean isKnown(String type) {
        return INSURANCE_VALUES.containsKey(type);
    }

    public static boolean isComponent(String type) {
        return "rune".equals(type) || "moonstone".equals(type);
    }

    public static int insuranceValue(String type) {
        return lookUp(INSURANCE_VALUES, type);
    }

    public static int basePremium(String type) {
        return lookUp(BASE_PREMIUMS, type);
    }

    private static int lookUp(Map<String, Integer> table, String type) {
        Integer value = table.get(type);
        if (value == null) {
            throw new ClaimOfficeException("unknown item type: " + type);
        }
        return value;
    }
}
